src/pages/Token/index.tsx

```diff
@@ -69,38 +71,77 @@ const TokenPage = () => {
     fetchToken();
   }, [push, fetchOffer, fetchToken, selectedAccount?.address, collectionId, id, token]);
 
+  const backClickHandler = useCallback((event: React.MouseEvent<HTMLAnchorElement>) => {
+    // if user opened token page with a direct link, redirect him to /market, otherwise redirect him back
+    if (history.length > 2) {
+      history.back();
+      event.preventDefault();
+    }
```

Comment:
missing redirect back to market if history is empty (`else history.push('')`)

Reply:
I would prefer to avoid such thing as "it is a urll, but somewhere inside we hide logic that makes it a button". Lets stick to it being just an element that we fully control instead of interfering link.